### ISSUE TITLE
Merge local news and tu news

### ISSUE THREAD

**Issue by dkehne** [digitalfabrik/integreat-app#2537]
As a user I want to have an overview of all news in one screen instead of clicking different news provider buttons. Due to the current planning of embeding Amal News as an additional content provider, we should work out a design which can handle n news providers.

## Acceptance Criteria:
- Local Push news should/could be hightlighted or priorized somehow because they might be more important/urgent that external Germany-wide news.

## Addtional information:

CMS issue: https://github.com/digitalfabrik/integreat-cms/issues/2833

[Design Proposal](https://www.figma.com/design/cA4F2MwHs2LNGviOWUjfE9/Integreat-Tickets-Frontend?node-id=4603-2945&t=zwFB1gFF4jl5urR9-1)

**Comment by sarahsporck** [comment]
Do we need filterbars? Is adding Amal News already discussed? So many questions ^^

**Comment by dkehne** [comment]
No Filterbar yet. A user just wants news, not filtering sources. 

Of course we should mark from which source the data comes, maybe in the same way we plan that with the external job and course data.

We decided to make a pilot and add Amal news in Maintal. It can be either instead of tünews or already together with it.

**Comment by sarahsporck** [comment]
Is there a due date for this pilot project?

**Comment by dkehne** [comment]
@sarahsporck - we have now received the API (same format as TüNews).

https://amalberlin.de/de/wp-json/wp/v2/posts
https://amalberlin.de/ua/wp-json/wp/v2/posts
https://amalberlin.de/fa/wp-json/wp/v2/posts
https://amalberlin.de/ar/wp-json/wp/v2/posts

I am not sure yet on which category id we should filter, but i will receive the info soon.

Coming back to your question: there is no due date offically. Completing the task in the first half of 2024 would be great, depending also on the ui-ux speed/capacities.

**Comment by steffenkleinle** [comment]
> @sarahsporck - we have now received the API (same format as TüNews).
> 
> https://amalberlin.de/de/wp-json/wp/v2/posts https://amalberlin.de/ua/wp-json/wp/v2/posts https://amalberlin.de/fa/wp-json/wp/v2/posts https://amalberlin.de/ar/wp-json/wp/v2/posts
> 
> I am not sure yet on which category id we should filter, but i will receive the info soon.
> 
> Coming back to your question: there is no due date offically. Completing the task in the first half of 2024 would be great, depending also on the ui-ux speed/capacities.

Two questions: Does it support pagination, i.e. just loading a specific amount of posts a time? Also, is there an endpoint for getting a single post by id?

**Comment by ztefanie** [comment]
It is a standard wordpress api, pagination and getting single posts is possible (just tested), check the docs here:
https://developer.wordpress.org/rest-api/reference/posts/#arguments

**Comment by dkehne** [comment]
The category taxonomy id for the non-regional news (we only should display) is 10884

**Comment by dkehne** [comment]
Is there any progress here or could someone from the team takeover?

**Comment by steffenkleinle** [comment]
No progress yet. We can easily implement amal news as an additional news tab/the only news source.

If we actually want to do it in one list we need a wrapper in the CMS scraping all the APIs (internal news, tünews, amal, ...) and returning everything from one endpoint (with pagination and so on).
That would be the best (i.e. cleanest) solution anyway and which is what we decided in the tech team what we want.

So we can do it quick in the frontend in another tab or we do it properly and handle that in the CMS. What do you prefer @dkehne? Then again its a question of prioritizing.

**Comment by steffenkleinle** [comment]
> @steffenkleinle it is not that urgent, because we plan to start the Amal-News-Embedding from January 2025.

Should work.

> So i would prefer to have one view (not the tabbing) where the entries are marked somehow so that the user can see what the source is (local integreat team, or external from TüNews/Amal or others). Could you hand it over to UI/UX to have a design?

Can you perhaps just add that information and maybe explain the purpose of the issue a little bit more in the issue description above (:arrow_up:)? Thanks.

> Do we need to have another issue for CMS or can we just handover that issue here somehow?

Yes, I think we need a separate CMS issue. I think we should get all news via one endpoint. Does this include tünews? I can create an issue.

**Comment by dkehne** [comment]
@steffenkleinle yes i can explain it a bit more. We want to have a generic approach to add different news-streams to the Integreat-plattform. We want to combine external news (e.g. from TüNews/Amal/....others) in one news feed with the local news. The source of the news (amal, tünews or the municipality itself via their cms-news-feature) should be visible in that list. 

I will move it to UI/UX nextup now...

**Comment by hauf-toni** [comment]
[Here](https://www.figma.com/design/cA4F2MwHs2LNGviOWUjfE9/Integreat-Tickets-Frontend?node-id=3223-3&t=vTND9H29Ats0Ov97-1) you can find the design proposal addressing the issue. Given the complexity of the previous discussions, I believe some aspects of the design might still need clarification. Please review the proposal and let me know if there are any points that remain unclear.

I recommend prioritizing local news in the feed (displaying them first). A quick research suggests that this function isn't heavily utilized by communities, so we likely won’t face an issue with the feed being overwhelmed by local news.

**Comment by dkehne** [comment]
@hauf-toni looks good. I have no more questions and like the idea of showing the local news first.
How can we ensure that the corporate design of the external news providers is somehow implemented to have a worth recognition (because they believe that this creates trustworthyness), e.g. if we would have tagesschau as a partner we would somehow implement their brand, wouldn't we?

**Comment by dkehne** [comment]
What is currently blocking this? @steffenkleinle

**Comment by steffenkleinle** [comment]
> What is currently blocking this? [@steffenkleinle](https://github.com/steffenkleinle)

We don't have a design yet and as mentioned before for me this is a CMS issue, see the issue linked in the description: https://github.com/digitalfabrik/integreat-cms/issues/2833

But perhaps we'd just need to have a bigger meeting on this including UI/UX and cms team.

**Comment by hauf-toni** [comment]
> We don't have a design yet and as mentioned before for me this is a CMS issue, see the issue linked in the description: [digitalfabrik/integreat-cms#2833](https://github.com/digitalfabrik/integreat-cms/issues/2833)
> 

The design is already done and just needs to be reviewed. I would recommending postponing this feature until after the implementation of the design + component system.

**Comment by osmers** [comment]
The CMS team already closed their issue to refactor Tü-News: https://github.com/digitalfabrik/integreat-cms/issues/4174
This means that all news will have the same structure in the API response to the app. @MizukiTemma is now adding Amal News as a news source. But in theory you can already start this issue @steffenkleinle . Let me know if there are any questions.
